codespell_lib/data/dictionary.txt

```diff
@@ -9773,7 +9776,7 @@ dordered->ordered
 dorment->dormant
 dorp->drop
 dosen't->doesn't
-dosen->doesn
+dosen->doesn't, dozen, doesn,
```

Comment:
I'm not sure if it makes sense to suggest `doesn` as a replacement?
Perhaps it's also worth adding `dose` as a suggested correction?

Reply:
peter wanted this to be included again. see [here](https://github.com/codespell-project/codespell/pull/1624#discussion_r471405252)

Reply:
Hmmm, I took that comment to mean "can you add `doesn` as an incorrectly-spelled word", but perhaps I misunderstood? @peternewman ?

Reply:
mmh makes sense. I will wait for reaction and fix it accordingly.

Reply:
I can't find the issue now, but essentially our default regex won't match doesn't (or dosen't) as the apostrophe isn't in the character list, it would match `dosen` as one word and `t` as another, so adding `doesn` as a suggestion would fix it, whereas just having doesn't would mean that `dosen't` would end up as `doesn't't` if that makes sense?

Someone else was suggesting stripping those words like `dosen't` out of the dictionary.

dose looks like a good suggestion anyway!

Reply:
If I check "dosen't" with my version of codespell it catches the "dosen't->doesn't" case and not this one.

can we throw it out?

Reply:
> our default regex won't match doesn't (or dosen't) as the apostrophe isn't in the character list, it would match dosen as one word and t as another

Ahhh, that makes sense now. Is there an issue for fixing that, is it a "known limitation" of codespell?
I guess in an ideal world codespell would be able to spot "This doesn have any typos" as having a typo, but "This doesn't have any typos" as not having a typo.

> Someone else was suggesting stripping those words like dosen't out of the dictionary.

Not sure if that'd help, but perhaps we should ensure that whenever we have cases like this, there's _always_ at least two suggested corrections, to prevent "This dosen have any typos" getting auto-corrected to "This doesn have any typos" ?

Reply:
Also, I wonder if `dosen->does` is a reasonable suggestion to add to the list?

Reply:
> Ahhh, that makes sense now. Is there an issue for fixing that, is it a "known limitation" of codespell?

It got discussed, but I can't find it now. Actually I think it might be #1482

I think you're damned either way, unless you run two passes. If you match `dosen't` as a typo, it means the apostrophe is a letter, so `'speling'` won't get caught, as that word (with the apostrophes) isn't in the dictionary (and that sort of thing is quite common in code).

> Not sure if that'd help, but perhaps we should ensure that whenever we have cases like this, there's _always_ at least two suggested corrections, to prevent "This dosen have any typos" getting auto-corrected to "This doesn have any typos" ?

That might make sense, but we shouldn't just add random corrections. I guess in this case if the corrections were `doesn` and `doesn't`, although that would trip up the current corrects to itself test.

I suspect the best solution is to move those errors into yet another dictionary file, and add a mechanism to propose the typo as itself for some dictionaries (it would make sense for names too).

> Also, I wonder if `dosen->does` is a reasonable suggestion to add to the list?

Then you're in a whole world of pain! A typo for doesn't getting corrected to does, or does't.

Reply:
Can we resolve this discussion in an issue/different MR? Because I don't see an easy fix right now and the initial purpose of this MR was to fix some falsely corrected words :-)

Reply:
Sorry for dragging this PR off-topic :wink:

Reply:
Yes definitely @sebweb3r , I'm always on-board with that if we've got something controversial so we can get the rest in.

Reply:
I removed doesn't and will do a new MR next week, after I had a more detailed look. It should be mergeable now.